org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -522,8 +522,9 @@ public void didChangeConfiguration(DidChangeConfigurationParams params) {
 			JavaLanguageServerPlugin.logException(e.getMessage(), e);
 		}
 		try {
-			boolean autoBuildChanged = ProjectsManager.setAutoBuilding(preferenceManager.getPreferences().isAutobuildEnabled());
-			if (jvmChanged || nullAnalysisOptionsUpdated) {
+			boolean isAutobuildEnabled = preferenceManager.getPreferences().isAutobuildEnabled();
+			boolean autoBuildChanged = ProjectsManager.setAutoBuilding(isAutobuildEnabled);
+			if (jvmChanged || nullAnalysisOptionsUpdated && isAutobuildEnabled) {
 				buildWorkspace(Either.forLeft(true));
 			} else if (autoBuildChanged) {
```

Comment:
it should be `else if (autobuildchanged && isAutobuildEnabled)` as well.  We need to buildWorkspace only when `autobuild` is changed from `false` to `true`.